### ISSUE TITLE
Comma in names breaks parsing RFC5322 entries

### ISSUE THREAD

**Issue by Manouchehri** [mat-sz/letterparser#16]
e.g. when I have this in my `To` header:

```
To: "last, first (test)" <example@test.mil>
```

This ends up getting parsed by letterparser like:
```
To: undefined <"last>
To: first (test) <example@test.mil>
```

**Comment by mat-sz** [comment]
@Manouchehri I can reproduce the bug, thank you for the report.

I'm hoping to release a fix soon.

**Comment by mat-sz** [comment]
@Manouchehri Resolved in 0.1.5.

Looking at this again, the codebase needs a serious refactor, but it should work for now.
